### PR TITLE
GSLA - handle DM01 and read product_type from gatts

### PR DESCRIPTION
**Pull request by lbesnard**

### CODE REVIEW

**Comment by lbesnard** [comment]
@ggalibert I can't remember the code now, but it was supposed to work

File: aodndata/gsla/handler.py [hunk 1]
```diff
@@ -78,10 +88,14 @@ def preprocess(self):
                 )
 
             netcdf_file = netcdf_collection[0]
+            # setting the path of the gz file with the gunqipped file
```
Comment (ggalibert) [inline aodndata/gsla/handler.py]: typo: gun**z**ipped

File: aodndata/gsla/handler.py [hunk 2]
```diff
@@ -78,10 +88,14 @@ def preprocess(self):
...
+            netcdf_file_gz.dest_path = self.dest_path(netcdf_file.src_path)
             # Nothing to do with *.nc. Talend can harvest *.nc.gz. Set to NO_ACTION
             netcdf_file.publish_type = PipelineFilePublishType.NO_ACTION
 
-            result_previous_version_creation_date = self.get_previous_version_creation_date(netcdf_file.name)
+            # we don't know the product type (DM00 or DM01) of the file already on s3 in order to deduce its path. So we
+            # to get the product type from the file in incoming
```
Comment (ggalibert) [inline aodndata/gsla/handler.py]: Fix sentence in comment. Missing "need" somewhere?

Comment (lbesnard) [reply aodndata/gsla/handler.py]: done

File: aodndata/gsla/handler.py [hunk 3]
```diff
@@ -48,14 +48,24 @@ def get_gsla_type(filepath):
...
 
     else:
         raise InvalidFileNameError(
             "file name: \"{filename}\" not matching regex to deduce dest_path".format(
                 filename=file_basename))
 
 
+def get_product_type(netcdf_path):
+    with Dataset(netcdf_path, mode='r') as nc_obj:
+        if hasattr(nc_obj, 'product_type'):
```
Comment (lwgordonimos) [inline aodndata/gsla/handler.py]: Can just invert this (EAFP, https://en.wikipedia.org/wiki/Python_syntax_and_semantics#Exceptions) to get rid of the redundant hasattr:

```python
try:
    return nc_obj.product_type
except AttributeError:
    raise InvalidInputFileError(
                "Expecting 'product_type' attribute in netCDF'{gzip}'".format(gzip=os.path.basename(netcdf_path))
            )
```

File: aodndata/gsla/handler.py [hunk 4]
```diff
@@ -148,13 +163,20 @@ def get_previous_version_creation_date(self, filepath):
...
+            dest_path_val = os.path.join(GSLA_PREFIX_PATH, gsla_type, file_basename)
 
         else:
             fields = get_pattern_subgroups_from_string(file_basename, GSLA_REGEX)
             gsla_year = datetime.strptime(fields['nc_time_cov_start'], '%Y%m%dT%H%M%SZ').year
-            return os.path.join(GSLA_PREFIX_PATH, gsla_type, str(gsla_year), file_basename)
+            dest_path_val = os.path.join(GSLA_PREFIX_PATH, gsla_type, str(gsla_year), file_basename)
+
+        # destination path should always be for nc.gz files. So we force it
+        if dest_path_val.endswith('.nc'):
```
Comment (lwgordonimos) [inline aodndata/gsla/handler.py]: If the if/else only controls a single value, you can just use a ternary style statement:

```python
return '{val}.gz'.format(val=dest_path_val) if dest_path_val.endswith('.nc') else dest_path_val
```

**Comment by lbesnard** [comment]
@lwgordonimos done

**Pull request merged**